pkg/server/options/authentication.go

```diff
@@ -107,6 +107,18 @@ func (s *AdminAuthentication) ApplyTo(config *genericapiserver.Config) (newToken
 	return newTokenOrEmpty, tokenHash, nil
 }
 
+func (s *AdminAuthentication) GetPushModeSyncerKubeconfig(config *genericapiserver.Config, newToken string, tokenHash []byte, externalAddress string, servingPort int) (*clientcmdapi.Config, error) {
+	externalCACert, _ := config.SecureServing.Cert.CurrentCertKeyContent()
+	externalKubeConfigHost := fmt.Sprintf("https://%s:%d", externalAddress, servingPort)
+
+	externalAdminUserName := "admin"
+	if newToken == "" {
+		return nil, fmt.Errorf("cannot create the PushModeSyncer kubeConfig with an empty token for the %s user", externalAdminUserName)
```

Comment:
which means this break on server restart.

Reply:
So, that's the reason it reads from the kubeconfig file...